tests/notificationprofile/V1/test_views.py

```diff
@@ -82,7 +84,164 @@ def test_notification_profile_can_properly_change_timeslot(self):
             self.notification_profile1.refresh_from_db()
         self.assertTrue(self.user1.notification_profiles.filter(pk=new_profile1_pk).exists())
         self.assertEqual(self.user1.notification_profiles.get(pk=new_profile1_pk).timeslot, self.timeslot2)
-        new_profile1_path = reverse("v1:notification-profile:notificationprofile-detail", args=[new_profile1_pk])
+        new_profile1_path = f"/api/v1/notificationprofiles/{new_profile1_pk}/"
         self.assertEqual(self.user1_rest_client.get(new_profile1_path).status_code, status.HTTP_200_OK)
 
-    # TODO: test more endpoints
+    def test_get_notification_profile_by_pk(self):
+        profile_pk = self.notification_profile1.pk
+        response = self.user1_rest_client.get(f"/api/v1/notificationprofiles/{profile_pk}/")
+        self.assertEqual(response.data["pk"], profile_pk)
+
+    def test_get_all_notification_profiles(self):
+        response = self.user1_rest_client.get("/api/v1/notificationprofiles/")
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["pk"], self.notification_profile1.pk)
+
+    def test_can_create_new_notification_profile_with_correct_values(self):
+        response = self.user1_rest_client.post(
+            "/api/v1/notificationprofiles/",
+            {
+                "timeslot": self.timeslot2.pk,
+                "filters": [f.pk for f in self.notification_profile1.filters.all()],
+                "media": self.media,
+                "phone_number": self.phone_number,
+                "active": self.notification_profile1.active,
+            },
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertTrue(NotificationProfile.objects.filter(pk=response.data.get("pk")))
+
+    def test_created_notificaton_profile_has_correct_media(self):
+        response = self.user1_rest_client.post(
+            "/api/v1/notificationprofiles/",
+            {
+                "timeslot": self.timeslot2.pk,
+                "filters": [f.pk for f in self.notification_profile1.filters.all()],
+                "media": self.media,
+                "phone_number": self.phone_number,
+                "active": self.notification_profile1.active,
+            },
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data["media"], self.media)
+
+    def test_can_delete_notification_profile(self):
+        profile_pk = self.notification_profile1.pk
+        response = self.user1_rest_client.delete(f"/api/v1/notificationprofiles/{profile_pk}/")
+        self.assertEqual(response.status_code, 204)
+        self.assertFalse(NotificationProfile.objects.filter(pk=profile_pk).first())
```

Comment:
`first` works, but `exists` makes it a little bit clearer
```suggestion
        self.assertFalse(NotificationProfile.objects.filter(pk=profile_pk).exists())
```